### PR TITLE
Add 'Metric Api' to Appdynamics Plugin (#144)

### PR DESCRIPTION
**Pull request by tgass**
Resolves issue #144

### CODE REVIEW

File: zmon_worker_monitor/builtins/plugins/appdynamics.py [hunk 1]
```diff
@@ -116,9 +116,92 @@ def __get_search_q(self, q, duration):
...
+        :param rollup: By default, the values of the returned metrics are rolled up into a single data point
+                       (rollup=true). To get separate results for all values within the time range, set the
+                       rollup parameter to false.
+        :type rollup: bool
+
+        :return: metric values for a metric
+        :rtype: list
+        """
+        try:
+            if metric_path is None:
```
Comment (mohabusama) [inline zmon_worker_monitor/builtins/plugins/appdynamics.py]: Is '' considered a valid ``metric_path``?

File: tests/test_appdynamics.py [hunk 2]
```diff
@@ -239,13 +319,25 @@ def test_appdynamics_healthrule_violations_errors(monkeypatch, fx_exception):
...
 
-    url = 'https://appdynamics'
-    application = 'App 1'
+    cli = AppdynamicsWrapper(URL, username=USER, password=PASS)
+
+    with pytest.raises(raised):
+        cli.healthrule_violations(APPLICATION, time_range_type=BEFORE_NOW, duration_in_mins=5)
+
+
+def test_appdynamics_metric_data_errors(monkeypatch, fx_exception):
```
Comment (mohabusama) [inline tests/test_appdynamics.py]: I suggest adding another test case for invalid kwargs. That way we can assert kwargs are validated in ``metric_data`` and easily detect regressions. Similar to ``test_appdynamics_healthrule_violations_kwargs_error``

**Comment by hjacobs** [comment]
I'm not familiar with AppDynamics API, but LGTM

**Comment by mohabusama** [comment]
👍

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**